### DISCUSSION TITLE
Issue with limiting case 1

### DISCUSSION THREAD

**Discussion by dkg-29** [Part-II-Computational-Physics/.github#64] [Q&A]
My plot does not seem to be working for limiting case 2.

<img width="546" alt="Screenshot 2023-03-08 230142" src="https://user-images.githubusercontent.com/123376173/223871834-b36224e2-6787-4650-a141-5ba5cc0fb218.png">

I don't think the error is related to the times selected as I find that M seems to tend to ~0.5 (but works well for all spins up or down).

<img width="390" alt="Screenshot 2023-03-08 230421" src="https://user-images.githubusercontent.com/123376173/223872406-e20ac3e8-9d02-4f01-b163-7acb3aa1a945.png">

I can't seem to find any error in the code.
<img width="919" alt="Screenshot 2023-03-08 231804" src="https://user-images.githubusercontent.com/123376173/223874571-79cca007-db0b-4de1-9bfb-81f7b9b212d0.png">

**Comment by DLHvdH** [comment]
Did you check that the computation of the magnetisation is correct? If you generate an all +1 lattice, the resulting magnetisation should be 1. Vice-versa for an all -1 lattice. You might also want to consider a slightly larger lattice (20 or 25), you don't have to go to such high time steps to get decent magnetisations. It's better though to average over a few more time steps.

**Reply by dkg-29** [reply]
The lattice generation and the computation of magnetisation seem to work. I am getting the same error with 20 steps.

<img width="641" alt="image" src="https://user-images.githubusercontent.com/123376173/224149271-fb560e42-5017-42d2-9c82-50c1aa7264a5.png">

**Reply by Star-Ruby** [reply]
For your random lattice, make sure the spins only take random values of either -1 or 1, and not random values between -1 and 1.

**Reply by dkg-29** [reply]
Everything works now - thanks!
